src/js/airplay/AirPlayButton.ts

```diff
@@ -0,0 +1,188 @@
+import videojs, { ComponentOptions } from '@silvermine/video.js';
+import type { VideoJsPlayer } from '../../../@types/videojs';
+import type { RemotePlaybackAvailabilityEvent } from '../../../@types/remote-playback';
+import { LOG_MESSAGES } from '../constants/log-messages';
+import { AVAILABILITY_STATES, EVENTS } from '../constants/remote-playback';
+import { getMediaElement } from '../lib/get-media-element';
+import { checkClientRemotePlaybackSupport, RemotePlaybackPlugin } from '../RemotePlaybackPlugin';
+import { checkClientAirPlaySupport } from './AirPlayManager';
+
+const CSS_CLASSES = {
+   AIRPLAY_BUTTON: 'vjs-airplay-button',
+   AIRPLAY_BUTTON_LARGE: 'vjs-airplay-button-lg',
+   AIRPLAY_BUTTON_LABEL: 'vjs-airplay-button-label',
+   AIRPLAY_CONNECTED: 'vjs-airplay-connected',
+   AIRPLAY_CONNECTING: 'vjs-airplay-connecting',
+   ICON_PLACEHOLDER: 'vjs-icon-placeholder',
+   CONTROL_TEXT: 'vjs-control-text',
+} as const;
+
+export const COMPONENT_NAMES = {
+   AIRPLAY_BUTTON: 'AirPlayButton',
+   CONTROL_BAR: 'controlBar',
+   FULLSCREEN_TOGGLE: 'fullscreenToggle',
+} as const;
+
+export interface AirPlayButtonOptions extends ComponentOptions {
+   addAirPlayLabelToButton?: boolean;
+}
+
+const Button = videojs.getComponent('Button');
+
+/*
+ * AirPlay control button for the Video.js control bar.
+ *
+ * Responsibilities:
+ * - Render a clickable control that asks the RemotePlayback plugin's
+ *   AirPlayManager to open the AirPlay device picker.
+ * - Decide when the button should be visible by listening to
+ *   availability events from both the plugin and the underlying
+ *   Remote Playback API.
+ * - Update its CSS classes when AirPlay is connecting / connected /
+ *   disconnected so the UI clearly reflects the current state.
+ */
+export class AirPlayButton extends Button {
+   private readonly _player: VideoJsPlayer;
+   private _labelEl?: HTMLSpanElement;
+
+   public constructor(player: VideoJsPlayer, options: AirPlayButtonOptions = {}) {
+      super(player, options);
+
+      this._player = player;
+
+      this._initializeButton(options);
+      this._setupEventListeners();
```

Comment:
Can we use the same consistent approach as in AirPlayManager with an `_initialize` method, and that method will contain all custom setup..

```suggestion
      super(player, options);
      this._player = player;
      this._initialize(options);
```